### PR TITLE
Pan to bounds instead of entity

### PR DESCRIPTION
**Pull request by briandaviddavidson**
Ticket: #624 

Expectation: When going into a pair of features, both should be visible on screen.

### CODE REVIEW

**Comment by briandaviddavidson** [comment]
jenkins please run cucumber

**Comment by rmrice** [comment]
@briandaviddavidson and I tested this on my machine and it works + passes cucumber tests.

**Comment by briandaviddavidson** [comment]
@brianhatchl Could this be an issue with my local vagrant box?

**Comment by brianhatchl** [comment]
@briandaviddavidson I know this is late, but you should not have to rebuild your entire vagrant vm.

Next time just try restarting tomcat:
`sudo service tomcat6 restart`

**Comment by brianhatchl** [comment]
I'm going to run the failing cucumber test locally.  But if it's not clear, here are the steps to prepare the PR code to be tested locally (from $HOOT_HOME on the vm host):

```
cd hoot-ui
git checkout 624
cd ..
vagrant ssh
cd hoot
source SetupEnv.sh
sudo -u tomcat6 scripts/vagrantDeployTomcat.sh
```

**Comment by brianhatchl** [comment]
This is failing for me locally, on the same step, but with a different error than the jenkins job:
```
vagrant@vagrant-ubuntu-trusty-64:~/hoot/test-files/ui$ xvfb-run --server-args="-screen 0, 1024x768x24" cucumber --format progress --strict features/resolve_all_reviews.feature
/home/vagrant/.rvm/gems/ruby-2.3.0/gems/selenium-webdriver-2.53.4/lib/selenium/webdriver/common/platform.rb:166: warning: Insecure world writable dir /home/vagrant/hadoop/bin/ in PATH, mode 040777
................F--------------FF

(::) failed steps (::)

undefined method `map' for nil:NilClass
Did you mean?  tap (NoMethodError)
./features/step_definitions/custom_steps.rb:61:in `/^I should see "([^"]*)"$/'
features/resolve_all_reviews.feature:20:in `Then I should see "Complete Review"'

Failing Scenarios:
cucumber features/resolve_all_reviews.feature:3 # Scenario: I can conflate the DC GIS data and resolve all reviews at once

1 scenario (1 failed)
31 steps (1 failed, 14 skipped, 16 passed)
1m26.277s
```

**Comment by brianhatchl** [comment]
I tried the cuke feature steps manually and got this error in the alert div:
```
TypeError: Cannot read property 'toString' of undefined
```
Unfortunately, the alert div seems to eat the js exception as it's not appearing in the dev console.

**Comment by brianhatchl** [comment]
The review features are selected, but the map is not zooming to them due to this error.
screenshot from 2016-08-08 15 13 33

File: js/hoot/control/conflicts/map/FeatureHighlighter.js [hunk 1]
```diff
@@ -151,7 +151,8 @@ Hoot.control.conflicts.map.featurehighlighter = function (context)
         _parent().reviewIds.push(relId);
         _parent().info.metadata.updateMeta(null);
         if(panToId && panTo) {
-            _parent().map.featureNavigator.panToEntity(context.entity(panToId), true);
+            var bounds = feature.loc.toString() + ',' + againstFeature.loc.toString();
```
Comment (brianhatchl) [inline js/hoot/control/conflicts/map/FeatureHighlighter.js]: Found the cause of the cuke error.  This won't work with features that are ways.  They don't have a "loc" property.

**Comment by brianhatchl** [comment]
I seem to remember working a long time ago on code that would make sure the review zoomed to the min bbox of the review features.  I'll update this with a reference to it when I can find it.

**Comment by brianhatchl** [comment]
Here is the commit, https://github.com/ngageoint/hootenanny-ui/commit/bb04cbe905a9f11b4b7326fadcf8f2394bc5181a
but it appears this behavior got refactored out of existence at some point.

**Comment by briandaviddavidson** [comment]
@brianhatchl Would it make sense to do separate functions to handle the different features when they go to the [`panToBounds` function](https://github.com/ngageoint/hootenanny-ui/blob/d5f91fe9931d2dbc0a13e382acb5416f82267582/js/hoot/control/conflicts/map/FeatureNavigator.js#L19) or do you think there is a better way to handle it?

**Comment by brianhatchl** [comment]
It looks like the highlightLayer method specifically deals with a pair of review features, so I would just want it to call a panToBounds method with the extent of the two features.

Or a new convenience method could be added panToEntities, that accepted an array of entity ids and then calculated the extent of those and zoomed to it.

**Comment by brianhatchl** [comment]
Seems like there might be a better syntax than using panToBounds.

```
map.centerZoom(extent.center(), map.extentZoom(extent));
```

You can just extend the extent of feature with review against feature and call this.  panToBounds can probably be removed entirely.

**Comment by briandaviddavidson** [comment]
Wouldn't extent in this case be the entire collection of features?

**Comment by briandaviddavidson** [comment]
test this please

**Comment by brianhatchl** [comment]
jenkins test this please

**Comment by brianhatchl** [comment]
npm test failure is caused by https://github.com/Medium/phantomjs/issues/599
will wait to see if it gets resolved upstream, but still run `npm test` locally before pushing commits please.

**Comment by jcp2236** [comment]
Passed local npm test on my machine. The offset error caused by expanding the sidebar affecting the radial menu, review bubbles, etc is also affecting the centering of the review nodes here. A fix should be coming soon for that issue.

**Comment by rmrice** [comment]
npm test works locally, failing on cucumber test

**Comment by brianhatchl** [comment]
run cucumber

**Comment by sputnam** [comment]
run cucumber

**Pull request merged**